### PR TITLE
Add shift api skeleton for all endpoints and validation

### PR DESCRIPTION
**Pull request by rupaltotale**

### CODE REVIEW

File: test-app/apps/backend-v2/src/app/shift/shift.dto.ts [hunk 1]
```diff
@@ -1,4 +1,21 @@
...
+import { datePreprocessSchema, laterDateAllowEqual } from "../utils";
+
+const BaseShift = z.object({
+  start: datePreprocessSchema(),
+  end: datePreprocessSchema(),
+  maxVolunteers: z.number().min(1),
+  title: z.string().min(1),
+  description: z.union([z.string().min(1), z.undefined()]),
+  eventAdmin: z.string(),
+});
```
Comment (bglossner) [inline test-app/apps/backend-v2/src/app/shift/shift.dto.ts]: Is this not in the shared package? Hoping we can avoid redefining things.

File: test-app/apps/backend-v2/src/app/utils/date.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import { z, ZodString, ZodError } from "nestjs-zod/z";
+
```
Comment (bglossner) [inline test-app/apps/backend-v2/src/app/utils/date.ts]: A lot of this also looks copied. Can it just be imported from the shared package instead?

**Review by bglossner — CHANGES_REQUESTED** [review]

File: test-app/apps/backend-v2/src/app/shift/shift.dto.ts (hunk 1 above)
Comment (rupaltotale) [reply test-app/apps/backend-v2/src/app/shift/shift.dto.ts]: Yep updated the shared package implementation to be compatible here

File: test-app/apps/backend-v2/src/app/utils/date.ts (hunk 2 above)
Comment (rupaltotale) [reply test-app/apps/backend-v2/src/app/utils/date.ts]: Same comment

**Pull request closed without merging**